channeldb/invoices.go

```diff
@@ -2384,14 +2427,20 @@ func (d *DB) cancelInvoice(invoices kvdb.RwBucket, invoiceNum []byte,
 		}
 	}
 
-	err = d.serializeAndStoreInvoice(invoices, invoiceNum, invoice)
+	err = d.cancelInvoiceStoreUpdate(invoices, invoiceNum, invoice)
```

Comment:
Eg of my ideas above:
  * This no longer takes the invoices directly. Instead just an `InvoiceUpdater` is passed in, which likely takes an abstract DB transaction. 
  * The updates are applied to that invoice pending state: `inv.UpdateState(Cancel)`, `inv.UpdateAllInvoiceStates(Cancel)`, then applied at the end `inv.Comit()`.

Reply:
Resolving this as the `InvoiceUpdater` is introduced later on.